File: app/src/main/java/com/example/foodplanner/presentation/onboarding/model/OnboardingItem.java
package com.example.foodplanner.presentation.onboarding.model;

public class OnboardingItem {
    
    private final String title;
    private final String description;
    private final int lottieRes;
    private final float maxProgress;

    public OnboardingItem(String title, String description, int lottieRes) {
        this(title, description, lottieRes, 1.0f);
    }

    public OnboardingItem(String title, String description, int lottieRes, float maxProgress) {
        this.title = title;
        this.description = description;
        this.lottieRes = lottieRes;
        this.maxProgress = maxProgress;
    }

    public String getTitle() {
        return title;
    }

    public String getDescription() {
        return description;
    }

    public int getLottieRes() {
        return lottieRes;
    }

    public float getMaxProgress() {
        return maxProgress;
    }
}

File: app/src/main/java/com/example/foodplanner/presentation/onboarding/OnboardingItem.kt
package com.example.foodplanner.presentation.onboarding

import androidx.annotation.RawRes

/**
 * Data class representing a single onboarding page.
 * 
 * @param title The main title text displayed on the page
 * @param description The description text providing more details
 * @param lottieRes The Lottie animation resource ID (from raw folder)
 */
data class OnboardingItem(
    val title: String,
    val description: String,
    @RawRes val lottieRes: Int
)
